src/renderer/components/+extensions/extensions.tsx

```diff
@@ -278,14 +322,30 @@ export class Extensions extends React.Component {
   }
 
   async uninstallExtension(extension: InstalledExtension) {
-    const extensionName = extensionDisplayName(extension.manifest.name, extension.manifest.version);
+    const displayName = extensionDisplayName(extension.manifest.name, extension.manifest.version);
 
     try {
+      this.setState(state => ({
+        ...state,
+        extensionState: {
+          ...state.extensionState,
+          [extension.id]: {
+            state: "uninstalling",
+            displayName
+          }
+        }
+      }));
```

Comment:
We use `mobx` instead of `setState`. I think that this easily could be an `observable.Map<string, {...}>` instead.

Reply:
I think local state (e.g. setState) is fine for component local state that should not be accessed outside. Our application is already using local state in various components.

Reply:
mobx manages local state. Those are mostly older components.

Reply:
Well, MobX _can_ manage local state, but that's a design choice. I'll convert it to MobX then.

Reply:
Converted